AwsCryptographyPrimitives/runtimes/go/ImplementationFromDafny-go/RSAEncryption/externs.go

```diff
@@ -90,30 +91,28 @@ func (CompanionStruct_Default___) EncryptExtern(paddingMode AwsCryptographyPrimi
 		return Wrappers.Companion_Result_.Create_Success_(dafny.SeqOfBytes(encryptedBytes))
 	}
 
-	hash1, err := getNativePaddingMode(paddingMode)
+	hash1, err := getNativeHashAlgorithm(paddingMode)
 	if err != nil {
 		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(err.Error())...)))
 	}
-	encryptedBytes, err := rsa.EncryptOAEP(crypto.Hash(hash1).New(), rand.Reader, publicKey.(*rsa.PublicKey), dafny.ToByteArray(plainText), nil)
+	encryptedBytes, err := rsa.EncryptOAEP(hash1.New(), rand.Reader, publicKey.(*rsa.PublicKey), dafny.ToByteArray(plainText), nil)
 	if err != nil {
 		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(err.Error())...)))
 	}
 	return Wrappers.Companion_Result_.Create_Success_(dafny.SeqOfBytes(encryptedBytes))
 }
 
-func getNativePaddingMode(mode AwsCryptographyPrimitivesTypes.RSAPaddingMode) (x509.SignatureAlgorithm, error) {
+func getNativeHashAlgorithm(mode AwsCryptographyPrimitivesTypes.RSAPaddingMode) (crypto.Hash, error) {
 	switch mode {
-	case AwsCryptographyPrimitivesTypes.Companion_RSAPaddingMode_.Create_PKCS1_():
```

Comment:
The case for `PKCS1` is already handled in line 85 (encrypt) and line 56 (decrypt). If we get `PKCS1` here, this is an error.

Reply:
Also, PKCS1 does not require hashing and returning error makes more sense in this function